hamocc/mo_bgcmean.F90

```diff
@@ -5,2354 +5,2216 @@
+    !
+    ! Arguments
+    real,    intent(in)    :: pddpo(idm,jdm,kdm)
+    integer, intent(in)    :: kin
+    integer, intent(inout) :: ind1(idm,jdm)
+    integer, intent(inout) :: ind2(idm,jdm)
+    real,    intent(inout) :: weights(idm,jdm,ddm)
+    !
+    ! Local variables
+    ! TODO: why do the following have save attributes?
+    integer         :: d,i,j,k,l
+    real,    save   :: dlev(idm,jdm,ddm)
+    real,    save   :: ztop(idm,jdm,kdm)
+    real,    save   :: zbot(idm,jdm,kdm)
+    logical, save   :: iniflg=.true.
+    real, parameter :: eps=1e-10
+    !
+    ! --- Adjust bounds of levitus levels according to model bathymetry
+    if (iniflg) THEN
+      do d=1,ddm
+        !$OMP PARALLEL DO PRIVATE(l,i)
+        do j=1,jj
+          do l=1,isp(j)
+            do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+              dlev(i,j,d)=max(eps, min(pbath(i,j),depthslev_bnds(2,d))-depthslev_bnds(1,d))
+            enddo
+          enddo
+        enddo
+        !$OMP END PARALLEL DO
+      enddo
+      iniflg=.false.
+    endif
+    !
+    ! --- Compute top and bottom depths of density layers
+    if (kin.EQ.1) THEN
+      !$OMP PARALLEL DO PRIVATE(l,i)
+      do j=1,jj
+        do l=1,isp(j)
+          do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+            zbot(i,j,1)=pddpo(i,j,1)
+          enddo
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+      do k=2,kk
+        !$OMP PARALLEL DO PRIVATE(l,i)
+        do j=1,jj
+          do l=1,isp(j)
+            do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+              zbot(i,j,k)=zbot(i,j,k-1)+pddpo(i,j,k)
+            enddo
+          enddo
+        enddo
+        !$OMP END PARALLEL DO
+      enddo
+      !$OMP PARALLEL DO PRIVATE(l,i)
+      do j=1,jj
+        do l=1,isp(j)
+          do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+            zbot(i,j,1)=zbot(i,j,1)*pbath(i,j)/zbot(i,j,kk)
+            ztop(i,j,1)=0.
+            ind1(i,j)=1
+          enddo
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+      do k=2,kk
+        !$OMP PARALLEL DO PRIVATE(l,i)
+        do j=1,jj
+          do l=1,isp(j)
+            do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+              zbot(i,j,k)=zbot(i,j,k)*pbath(i,j)/zbot(i,j,kk)
+              ztop(i,j,k)=zbot(i,j,k-1)
+            enddo
+          enddo
+        enddo
+        !$OMP END PARALLEL DO
+      enddo
+    endif
+    !
+    !--- Compute interpolation weights
+    !$OMP PARALLEL DO PRIVATE(l,i,d)
+    do j=1,jj
+      do l=1,isp(j)
+        do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+          ind2(i,j)=0
+          if (pddpo(i,j,kin).GT.eps) THEN
+            do d=ind1(i,j),ddm
+              if (depthslev_bnds(2,d).LE.ztop(i,j,kin)) THEN
+                ind1(i,j)=d+1
+                CYCLE
+              elseif (depthslev_bnds(1,d).GE.zbot(i,j,kin)) THEN
+                EXIT
+              endif
+              ind2(i,j)=d
+              weights(i,j,d)=(min(zbot(i,j,kin), depthslev_bnds(2,d)) - &
```

Comment:
This change in line break feels odd - there was a clear meaning before (`depthlev_bnds(2,d) - ` should be placed in the next line)

Reply:
Done. Thanks for pointing this out.